### ISSUE TITLE
Seeing “Error executing command: begin 0, end -1, length 160” errors

### ISSUE THREAD

**Issue by mjpieters** [SOBotics/Guttenberg#204]
See [the chat logs](https://chat.stackoverflow.com/transcript/message/48467290#48467290), where I make repeated attempts to get Gut to give me more info on a serial plagiarist.

I have no idea where that error originates from; the full `-log` output naturally terminates just *before* the problematic post.

To improve robustness, perhaps the [per post loop body](https://github.com/SOBotics/Guttenberg/blob/d888c0d1ebe4cc22463d583958e9c395c9e79292/src/main/java/org/sobotics/guttenberg/commands/CheckUser.java#L120-L155) could be protected by a `try {}` block that reports on individual problems and continues with the next post?

**Comment by FelixSFD** [comment]
```
java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 126
        at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
        at java.base/java.lang.String.substring(String.java:1874)
        at org.sobotics.guttenberg.search.UserAnswer.getSearchString(UserAnswer.java:162)
        at org.sobotics.guttenberg.commands.CheckUser.execute(CheckUser.java:131)
        at org.sobotics.guttenberg.commandlists.SoBoticsCommandsList.mention(SoBoticsCommandsList.java:94)
        at org.sobotics.guttenberg.roomdata.SOBoticsWorkshopChatRoom.lambda$getMention$0(SOBoticsWorkshopChatRoo$
        at org.sobotics.chatexchange.chat.Room.lambda$null$6(Room.java:231)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```


Probably this piece of code:
```
if (content != null && content.length() > 100 && content.indexOf(' ', 90)>0) {
  return new UserAnswerLine(LineType.TEXT, content.substring(0, content.indexOf(' ', 90)));
}
```

But I don't understand how `content.indexOf(' ', 90)` can be `-1` in the second line. :-/

**Comment by jdd-software** [comment]
@FelixSFD maybe this version is not deployed, anyway for sure a try catch (Throwable e), on Martjin block would skip the answer, not the best thing to do but well it would work :)

**Comment by mjpieters** [comment]
> But I don't understand how `content.indexOf(' ', 90)` can be `-1` in the second line. :-/

jdd is correct, `@gut status` shows it is running version 1.2.5 and the [check for a space](https://github.com/SOBotics/Guttenberg/commit/033fe98066d87cc4a5a08ad43efa7e9b53f2d249) was added in 1.2.6.

**Comment by FelixSFD** [comment]
Should be fixed in 1.2.7 via #199
